src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/LanguageServerFeatureOptions.cs

```diff
@@ -14,5 +14,17 @@ internal abstract class LanguageServerFeatureOptions
         public abstract string HtmlVirtualDocumentSuffix { get; }
 
         public abstract bool SingleServerCompletionSupport { get; }
+
+        public string GetRazorCSharpFilePath(string razorFilePath) => razorFilePath + CSharpVirtualDocumentSuffix;
+        
+        public string GetRazorHtmlFilePath(string razorFilePath) => razorFilePath + HtmlVirtualDocumentSuffix;
+
+        public string GetRazorFilePath(string filePath)
+        {
+            filePath = filePath.Replace(CSharpVirtualDocumentSuffix, string.Empty);
```

Comment:
Super Nit: I assume that string.Replace doesn't actually allocate a new string if there's nothing to replace?

Reply:
I'd imagine it'd be smart but not sure 😄